service/deployer/deployer.go

```diff
@@ -50,12 +59,21 @@ func New(config Config) (Deployer, error) {
 		return nil, microerror.MaskAnyf(invalidConfigError, "logger must not be empty")
 	}
 
+	// Settings.
+	if config.Flag == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "flag must not be empty")
+	}
+	if config.Viper == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "viper must not be empty")
+	}
```

Comment:
It would be good to validate that there are none nil clients provided. Otherwise this code can panic at runtime when not properly configured.